plugin/pkg/admission/serviceplan/defaultserviceplan/admission.go

```diff
@@ -123,7 +123,7 @@ func (d *defaultServicePlan) Admit(a admission.Attributes) error {
 	p := plans[0]
 	glog.V(4).Infof("Using default plan %q for Service Class %q for instance %s",
 		p.Name, sc.Name, instance.Name)
-	instance.Spec.PlanName = p.Name
+	instance.Spec.ExternalServicePlanName = p.Name
```

Comment:
This should be `instance.Spec.ExternalServicePlanName = p.Spec.ExternalName` I believe.

Reply:
In `plugin/pkg/admission/serviceplan/defaultserviceplan/admission_test.go`, we should also set the names in `func newServicePlans(count uint)` to not be the same as the ExternalName so that we don't get any false positives.

Reply:
Great catch.